flag_engine/segments/evaluator.py

```diff
@@ -156,52 +170,61 @@ def _evaluate_modulo(
-        if isinstance(trait_value, int) and not any(
-            trait_value is x for x in (False, True)
+        if isinstance(context_value, str):
+            return context_value in segment_value.split(",")
+        if isinstance(context_value, int) and not any(
+            context_value is x for x in (False, True)
         ):
-            return str(trait_value) in segment_value.split(",")
+            return str(context_value) in segment_value.split(",")
     return False
 
 
-def _trait_value_typed(
+def _context_value_typed(
     func: typing.Callable[..., bool],
 ) -> typing.Callable[[typing.Optional[str], TraitValue], bool]:
     @wraps(func)
     def inner(
         segment_value: typing.Optional[str],
-        trait_value: typing.Union[TraitValue, semver.Version],
+        context_value: typing.Union[TraitValue, semver.Version],
     ) -> bool:
         with suppress(TypeError, ValueError):
-            if isinstance(trait_value, str) and is_semver(segment_value):
-                trait_value = semver.Version.parse(
-                    trait_value,
+            if isinstance(context_value, str) and is_semver(segment_value):
+                context_value = semver.Version.parse(
+                    context_value,
                 )
-            match_value = get_casting_function(trait_value)(segment_value)
-            return func(trait_value, match_value)
+            match_value = get_casting_function(context_value)(segment_value)
+            return func(context_value, match_value)
         return False
 
     return inner
 
 
-MATCH_FUNCS_BY_OPERATOR: typing.Dict[
+MATCHERS_BY_OPERATOR: typing.Dict[
     ConditionOperator, typing.Callable[[typing.Optional[str], TraitValue], bool]
 ] = {
     constants.NOT_CONTAINS: _evaluate_not_contains,
     constants.REGEX: _evaluate_regex,
     constants.MODULO: _evaluate_modulo,
     constants.IN: _evaluate_in,
-    constants.EQUAL: _trait_value_typed(operator.eq),
-    constants.GREATER_THAN: _trait_value_typed(operator.gt),
-    constants.GREATER_THAN_INCLUSIVE: _trait_value_typed(operator.ge),
-    constants.LESS_THAN: _trait_value_typed(operator.lt),
-    constants.LESS_THAN_INCLUSIVE: _trait_value_typed(operator.le),
-    constants.NOT_EQUAL: _trait_value_typed(operator.ne),
-    constants.CONTAINS: _trait_value_typed(operator.contains),
+    constants.EQUAL: _context_value_typed(operator.eq),
+    constants.GREATER_THAN: _context_value_typed(operator.gt),
+    constants.GREATER_THAN_INCLUSIVE: _context_value_typed(operator.ge),
+    constants.LESS_THAN: _context_value_typed(operator.lt),
+    constants.LESS_THAN_INCLUSIVE: _context_value_typed(operator.le),
+    constants.NOT_EQUAL: _context_value_typed(operator.ne),
+    constants.CONTAINS: _context_value_typed(operator.contains),
+}
+
+
+CONTEXT_VALUE_GETTERS_BY_PROPERTY = {
+    "$.identity.identifier": itemgetter("identity", "identifier"),
+    "$.identity.key": itemgetter("identity", "key"),
```

Comment:
Should we add this one too? Right now it's using the flag to grab the context values. Even though we gain in terms of flexibility, i'm not convinced by the upside of doing so.
We lose in terms of validation and control.

Ideally the API provides the list (similar to what we do with permissions) even though it might be overkill.

In the meanwhile, I do think it would be better to rely on having the context values as constants in the frontend, have a flag to enable/disable as a whole and per context value:

`{ disabled: [$.identity.idenfier, xxx]}` for example